setup.py

```diff
@@ -191,36 +189,109 @@ def finalize_options(self):
+        if dest_folder.exists():
+            shutil.rmtree(dest_folder)
+        os.makedirs(dest_folder, exist_ok=True)
+        print(f'Functions Host folder is created in {dest_folder}')
+
+    def _extract_webhost_zip(self, version: str, src_zip: str, dest: str):
+        print(f'Extracting Functions Host from {src_zip}')
+
+        with zipfile.ZipFile(src_zip) as archive:
+            # We cannot simply use extractall(), as the archive
+            # contains Windows-style path names, which are not
+            # automatically converted into Unix-style paths, so
+            # extractall() will produce a flat directory with
+            # backslashes in file names.
+
+            for archive_name in archive.namelist():
+                prefix = f'azure-functions-host-{version}/'
+                if archive_name.startswith(prefix):
+                    sanitized_name = archive_name \
+                        .replace('\\', os.sep) \
+                        .replace(prefix, '')
+                    dest_filename = dest / sanitized_name
+                    zipinfo = archive.getinfo(archive_name)
+
+                    try:
+                        if not dest_filename.parent.exists():
+                            os.makedirs(dest_filename.parent, exist_ok=True)
+
+                        if zipinfo.is_dir():
+                            os.makedirs(dest_filename, exist_ok=True)
+                        else:
+                            with archive.open(archive_name) as src, \
+                                    open(dest_filename, 'wb') as dst:
+                                dst.write(src.read())
+                    except Exception as e:
+                        print(f'Failed to extract file {archive_name}'
+                              f': {e!r}', file=sys.stderr)
+                        sys.exit(1)
+
+        print(f'Functions Host is extracted into {dest}')
+
+    def _chmod_protobuf_generation_script(self, webhost_dir: pathlib.Path):
+        # This script is needed to set to executable in order to build the
+        # WebJobs.Script.Grpc project in Linux and MacOS
+        script_path = (
+            webhost_dir / 'src' / 'WebJobs.Script.Grpc' / 'generate_protos.sh'
+        )
+        if sys.platform != 'win32' and os.path.exists(script_path):
+            print('Change generate_protos.sh script permission')
+            os.chmod(script_path, 0o555)
+
+    def _compile_webhost(self, webhost_dir: pathlib.Path):
+        print(f'Compiling Functions Host from {webhost_dir}')
+
+        try:
+            subprocess.run(
+                args=['dotnet', 'build', 'WebJobs.Script.sln', '-o', 'bin'],
+                check=True,
+                cwd=str(webhost_dir),
+                stdout=sys.stdout, stderr=sys.stderr)
+        except Exception:
+            print(f"Failed to compile webhost in {webhost_dir}. "
+                  ".NET Core SDK is required to build the solution",
```

Comment:
nit; Just add `https://aka.ms/dotnet-download` link to the error, with a recommendation of which version to install.